services/src/main/java/org/keycloak/storage/UserStorageManager.java

```diff
@@ -161,10 +162,8 @@ protected void deleteInvalidUser(final RealmModel realm, final UserModel user) {
         Stream<UserModel> query(Object provider);
     }
 
-    protected Stream<UserModel> query(PaginatedQuery pagedQuery, RealmModel realm, int firstResult, int maxResults) {
-        if (maxResults == 0) return Stream.empty();
-        if (firstResult < 0) firstResult = 0;
-        if (maxResults < 0) maxResults = Integer.MAX_VALUE - 1;
+    protected Stream<UserModel> query(PaginatedQuery pagedQuery, RealmModel realm, Integer firstResult, Integer maxResults) {
+        if (maxResults != null && maxResults == 0) return Stream.empty();
```

Comment:
This line will make sure that empty stream is returned when `maxResult == 0`. Even without this check the `paginatedStream` function will add `.limit(0)` because there is `>= 0`. I am not sure I understand what you mean.

Reply:
You're right, my mistake, please ignore.